coda-src/venus/vproc_pioctl.cc

```diff
@@ -168,11 +168,17 @@ void vproc::do_ioctl(VenusFid *fid, unsigned char nr, struct ViceIoctl *data)
 			(void)k_Purge(fid, 1);
 
 			f = FSDB->Find(fid);
-			if (f != 0) {
-			    u.u_error = f->Flush();
-			    Recov_SetBound(DMFP);
-			    f = 0;
-			}
+			if (f == 0) break;
+
+            if (ISVASTRO(f) && ACTIVE(f)) {
```

Comment:
Doesn't f->Flush already do this?

Reply:
Not really. Currently, `f->Flush()` doesn't discard any data. However, it will remove the `fsobj` from the `FSDB` and, again, we don't want this for active VASTROs.